modules/extensions-mule-support/src/main/java/org/mule/runtime/module/extension/mule/internal/loader/parser/utils/AggregatedErrorsCharacteristic.java

```diff
@@ -78,10 +84,46 @@ public List<ErrorModelParser> apply(ComponentAstWithHierarchy componentAstWithHi
     private void handleOperationOtherThanRaiseError(ComponentAst operationAst, List<ErrorModelParser> models,
                                                     List<ComponentAst> hierarchy) {
       Optional<OperationModel> operationModel = operationAst.getModel(OperationModel.class);
-      operationModel.ifPresent(model -> model.getErrorModels().stream().map(MuleSdkErrorModelParser::new)
+      operationModel.ifPresent(model -> model.getErrorModels().stream()
+          .map(errorModel -> createErrorModelParser(errorModel, operationAst))
           .forEach(errorModelParser -> addParserAndMarkIfSuppressed(errorModelParser, models, hierarchy)));
     }
 
+    private MuleSdkErrorModelParser createErrorModelParser(ErrorModel errorModel, ComponentAst operationAst) {
+      final ComponentParameterAst errorMappingsParam = operationAst.getParameter(ERROR_MAPPINGS, ERROR_MAPPINGS_PARAMETER_NAME);
+      if (errorMappingsParam == null) {
+        return new MuleSdkErrorModelParser(errorModel);
+      }
+
+      Optional<List<ErrorMapping>> errorMappingsOpt = errorMappingsParam.<List<ErrorMapping>>getValue().getValue();
+      if (!errorMappingsOpt.isPresent()) {
+        return new MuleSdkErrorModelParser(errorModel);
+      }
+
+      List<ErrorMapping> errorMappings = errorMappingsOpt.get();
+      return applyMappingIsSomeMatches(errorModel, errorMappings);
+    }
+
+    private MuleSdkErrorModelParser applyMappingIsSomeMatches(ErrorModel errorModel, List<ErrorMapping> errorMappings) {
+      return errorMappings.stream()
+          .filter(errorMapping -> doesMappingSourceMatch(errorMapping, errorModel))
+          .map(Aggregator::buildParserFromTarget)
+          .findFirst()
+          .orElseGet(() -> new MuleSdkErrorModelParser(errorModel));
+    }
+
+    private static MuleSdkErrorModelParser buildParserFromTarget(ErrorMapping errorMapping) {
+      String mappingTargetAsString = errorMapping.getTarget();
+      ComponentIdentifier targetErrorId = parseErrorType(mappingTargetAsString, CORE_ERROR_NAMESPACE);
+      return new MuleSdkErrorModelParser(targetErrorId.getNamespace(), targetErrorId.getName(), null);
```

Comment:
If the error was from `CORE` (or eventually from the same extension), shouldn't we try to retain the parent information? here it seems we are always setting it to `null`.

Reply:
The target namespace in error-mapping might be:
1. The namespace "THIS": It's not allowed by a AST validation that prevents it to use namespaces from some extension. If you want to raise an error from such namespace, you need to use the operation:raise-error component
2. The namespace of some existing extension: It's not allowed by the same AST validation
3. The core namespace ("MULE"): It is allowed, but is there some core error that doesn't inherit from MULE:ANY directly? I can't find one
4. Some original super-cool namespace, as "SUPERCOOL": In this case, there isn't any syntax to make that error not-a-child of MULE:ANY

Reply:
3. I answer myself: Actually there are several core errors with a parent other than MULE:ANY. Working on that...

Reply:
Fixed